web/api/authorization.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+const anonymousRole = "anonymous"
+
+type authzInfo struct {
+	appID       string
+	tokenStr    string
+	resourceURI string
+	method      string
+}
+
+// Authorize checks if user has an access to the requested resource.
+// If error happens, writes it to ResponseWriter.
+// Also, writes an error on failed authorization.
+func (ar *Router) Authorize(w http.ResponseWriter, azi authzInfo) error {
```

Comment:
we don't have user->resource auth scheme

Reply:
It's a role->resource scheme essentially.